File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int sum=0;
        int maxi=0;
        int count =INT_MAX;
        for(int i=0;i<prices.size();i++){
            if(prices[i]<count){
                count=prices[i];
            }
            sum=prices[i]-count;
            maxi=max(sum,maxi);
    }
        return maxi;
    }
};

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        
        int n=prices.length;
        int temp=0;
        int sum=Integer.MAX_VALUE;
        int count=0;
        for(int i=0;i<n;i++){
            if(prices[i]<sum)
                sum=prices[i];
            temp=prices[i]-sum;
            count=Math.max(temp,count);
        }
        return count;
    }
}